### PR TITLE
Closes #1422 adlb ncictcv5@devel

### LINKED ISSUE #1422: Feature Request:  Create NCICTCAEv5 metadata (similar to NCICTCAEv4 implementation)
### Feature Idea

We released Lab grading in release 0.9, implementing NCICTCAEv4. We should now implement NCICTCAEv5, using same metadata structure.

### Relevant Input

See website for NCICTCAEv5 grading: https://ctep.cancer.gov/protocoldevelopment/electronic_applications/ctc.htm

### Relevant Output

Same structure as data("atoxgr_criteria_ctcv4")

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by millerg23**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: vignettes/lab_grading.Rmd [hunk 1]
```diff
@@ -26,16 +26,18 @@ Within the ADLB ADaM data set there is a concept of lab grading, where there is
 a set of criteria for particular lab tests that grade the severity or abnormality of 
 a lab value. The grades are from 0 to 4, where grade 0 can be viewed generally as a 
 “NORMAL” value. The higher the grade the more severe or more abnormal the lab value is.
-There are several sets of lab grading criteria, for the initial implementation of lab
-grading we will look at NCI-CTCAEv4. (In future releases `{admiral}` look to implement
-further grading criteria, for example NCI-CTCAEv5)
+There are several sets of lab grading criteria, currently {admiral} has implemented
```
Comment (bms63) [inline vignettes/lab_grading.Rmd:29]:
```suggestion
There are several sets of lab grading criteria, currently `{admiral}` has implemented
```

File: vignettes/lab_grading.Rmd [hunk 2]
```diff
@@ -26,16 +26,18 @@ Within the ADLB ADaM data set there is a concept of lab grading, where there is
...
+NCI-CTCAEv4 and NCI-CTCAEv5 grading criteria. (In future releases `{admiral}` may look
+to implement further grading criteria).
```
Comment (bms63) [inline vignettes/lab_grading.Rmd:31]:
```suggestion
NCI-CTCAEv4 and NCI-CTCAEv5 grading criteria. In future releases `{admiral}` may look
to implement further grading criteria.
```

File: vignettes/lab_grading.Rmd [hunk 3]
```diff
@@ -104,6 +106,21 @@ atoxgr_criteria_ctcv4 %>%
   )
 ```
 <br>
+
+<br>
+Likewise, the list of terms defined in the `{admiral}` metadata to implement NCI-CTCAEv5 is below:
+(Terms identical to NCI-CTCAEv4, except `Hyperglycemia`, `Hyperglycemia (Fasting)` and `Hypophosphatemia`)
+which are not present in NCI-CTCAEv5.
+<br>
```
Comment (bms63) [inline vignettes/lab_grading.Rmd:114]: Should we make a note of this in the function as well??

File: R/data.R [hunk 4]
```diff
@@ -67,3 +68,39 @@
...
+#'
+#' Note: Variables `SOC`, `TERM`, `Grade 1`, `Grade 2`,`Grade 3`,`Grade 4`,`Grade 5`, `Definition`
+#' are from the source document on NCI-CTC website defining the grading criteria.
+#' From these variables only 'TERM' is used in the {admiral} code, the rest are for information and
+#' tracability only.
+#'
+#' @author Gordon Miller
+#'
+#' @keywords datasets
+#' @family datasets
```
Comment (bms63) [inline R/data.R:105]: I think we should change this to `metadata` we recently added this as a keyword/family

Comment (bms63) [reply R/data.R:105]: image

File: R/data.R [hunk 5]
```diff
@@ -37,7 +37,8 @@
 #'
 #' @details
 #' This metadata has its origin in the ADLB Grading Spec Excel file which ships with `{admiral}`
-#' and can be accessed using `system.file("adlb_grading/adlb_grading_spec.xlsx", package = "admiral")`.
+#' and can be accessed using `system.file("adlb_grading/adlb_grading_spec.xlsx", package = "admiral")`
+#' in sheet = "NCICTCAEv4".
```
Comment (bms63) [inline R/data.R:41]: Can we include  the hyperlink to the NCI stuff in metadata and the function as well as hyperlink to the vignette?? 

image

File: R/data.R [hunk 6]
```diff
@@ -67,3 +68,39 @@
...
+#' - `VAR_CHECK`: List of variables required to implement lab grade criteria. Use to check against
+#'   input data.
+#' - `DIRECTION`: variable to hold the direction of the abnormality of a particular lab test
+#'   value. 'L' is for LOW values, 'H' is for HIGH values. Note: the variable is case insensitive.
+#' - `COMMENT`: Holds any information regarding rationale behind implementation of grading criteria.
+#'
+#' Note: Variables `SOC`, `TERM`, `Grade 1`, `Grade 2`,`Grade 3`,`Grade 4`,`Grade 5`, `Definition`
+#' are from the source document on NCI-CTC website defining the grading criteria.
+#' From these variables only 'TERM' is used in the {admiral} code, the rest are for information and
+#' tracability only.
```
Comment (bms63) [inline R/data.R:100]:
```suggestion
#' traceability only.
```

Comment (bms63) [reply R/data.R:100]: surprised the spell checker didn't flag this...wondering if it is not checking rmd files

File: R/data.R (hunk 5 above)
Comment (bms63) [reply R/data.R:41]: sorry i see it in the function - just in the metadata then??

**Comment by bms63** [comment]
Should this example have dummy subjects in it?  
image

File: R/data.R (hunk 4 above)
Comment (millerg23) [reply R/data.R]: Done.

File: R/data.R (hunk 5 above)
Comment (millerg23) [reply R/data.R:41]: Done, I think. Can you double-check I updated where you wanted?

File: vignettes/lab_grading.Rmd (hunk 3 above)
Comment (millerg23) [reply vignettes/lab_grading.Rmd:114]: What function? `derive_var_atoxgr_dir` ? I don't think so, I would expect users to look in the Vignette to find details of whats implemented.

**Comment by millerg23** [comment]
Think I resolved everying @bms63. Regarding example having subject IDs, I was trying to save the space for the variabes that have an impact on the grade. I could add subject IDs if you feel strongly about it.

**Comment by millerg23** [comment]
Hmm,  checks failing due to OCCDS vignette! Wil try merging in `devel` branch and see if it still fails.

**Comment by bms63** [comment]
> Think I resolved everying @bms63. Regarding example having subject IDs, I was trying to save the space for the variabes that have an impact on the grade. I could add subject IDs if you feel strongly about it.

That makes sense.  @thomas-neitmann any thoughts on not including subject ids?

**Review by yangc71 — APPROVED** [review]
Hi Gordon, I looked through the test examples where v5 differs from v4 and I think you did a great job. It's a lot of effort to write up all those test cases!

**Comment by millerg23** [comment]
@yangc71 thanks for looking into the unt tests, hope your eyes didn't go cross-ways ;-)
@bms63 just spotted ADLB template failing, I will take a look.

**Comment by millerg23** [comment]
@bms63 could yo review updates I made to ADLB template around the grades, ie NCICTCaev4/v5.
Thanks!

File: R/data.R (hunk 5 above)
Comment (bms63) [reply R/data.R:41]: yes

**Review by bms63 — APPROVED** [review]
Looks great!!  Thanks Gordon!!

**Pull request merged**